app/vmui/packages/vmui/src/pages/ExploreAlerts/ExploreRules.tsx

```diff
@@ -0,0 +1,266 @@
+
+const defaultTypesStr = getQueryStringValue("types", "") as string;
+const defaultTypes = defaultTypesStr.split("&").filter((rt) => rt) as string[];
+const defaultStatesStr = getQueryStringValue("states", "") as string;
+const defaultStates = defaultStatesStr.split("&").filter((s) => s) as string[];
+const defaultSearchInput = getQueryStringValue("search", "") as string;
+
+interface ExploreRulesProps {
+  typeFilter: string;
+}
+
+const ExploreRules: FC<ExploreRulesProps> = ({ typeFilter }) => {
+  const groupId = getQueryStringValue("group_id", "") as string;
+  const ruleId = getQueryStringValue("rule_id", "") as string;
+  const alertId = getQueryStringValue("alert_id", "") as string;
+
+  const [expandGroups, setExpandGroups] = useState<Set<string>>(
+    new Set<string>(),
+  );
+  const [expandRules, setExpandRules] = useState<Set<string>>(
+    new Set<string>(),
+  );
+  const [searchInput, setSearchInput] = useState(defaultSearchInput);
+  const [types, setTypes] = useState(defaultTypes);
+  const [states, setStates] = useState(defaultStates);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const { hash } = useLocation();
+
+  if (!hash && groupId && (ruleId || alertId)) {
+    setModalOpen(true);
+  } else {
+    setModalOpen(false);
+  }
+
+  useSetQueryParams({
+    types: types.join("&"),
+    states: states.join("&"),
+    search: searchInput,
+    group_id: groupId,
+    alert_id: alertId,
+    rule_id: ruleId,
+  });
+
+  const handleChangeSearch = (input: string) => {
+    if (!input) {
+      setSearchInput("");
+    } else {
+      setSearchInput(input);
+    }
+  };
+
+  const handleGroupChangeExpand = useCallback(
+    (id: string) => (value: boolean) => {
+      setExpandGroups((prev) => {
+        const newExpandGroups = new Set(prev);
+        if (value) {
+          newExpandGroups.add(id);
+        } else {
+          newExpandGroups.delete(id);
+        }
+        return newExpandGroups;
+      });
+    },
+    [],
+  );
+
+  const handleRuleChangeExpand = useCallback(
+    (id: string) => (value: boolean) => {
+      setExpandRules((prev) => {
+        const newExpandRules = new Set(prev);
+        if (value) {
+          newExpandRules.add(id);
+        } else {
+          newExpandRules.delete(id);
+        }
+        return newExpandRules;
+      });
+    },
+    [],
+  );
+
+  const handleChangeStates = (title: string) => {
+    setStates(getChanges(title, states));
+  };
+
+  const handleChangeTypes = (title: string) => {
+    setTypes(getChanges(title, types));
+  };
+
+  let noRuleFound = "No rules found!";
+  if (typeFilter === "alert") {
+    noRuleFound = "No alerting rules found!";
+  } else if (typeFilter === "record") {
+    noRuleFound = "No recording rules found!";
+  }
+
+  const handleClose = () => {
+    searchParams.delete("group_id");
+    searchParams.delete("rule_id");
+    searchParams.delete("alert_id");
```

Comment:
It's better to avoid mutating state objects, and do somethings like 
```suggestion
  const newParams = new URLSearchParams(searchParams);
  newParams.delete("group_id");
  newParams.delete("rule_id");
  newParams.delete("alert_id");
  setSearchParams(newParams);
```